lib/optimize/SideEffectsFlagPlugin.js

```diff
@@ -117,5 +116,34 @@ class SideEffectsFlagPlugin {
 			});
 		});
 	}
+
+	static moduleHasSideEffects(modulePath, flagValue) {
+		switch(typeof flagValue) {
+			case "boolean":
+				return flagValue;
+			case "string":
+				{
+					if(flagValue === "false" || flagValue === "true") {
+						return flagValue === "false" ? (/false.*$/).test(modulePath) : true;
+					}
+
+					if(flagValue === modulePath) {
+						return true;
+					}
+
+					let matchesRegex;
+					try {
+						matchesRegex = (new RegExp(flagValue, "i")).test(modulePath);
+					} catch(e) {}
+					return matchesRegex || globToRegExp(flagValue).test(modulePath);
+				}
+			case "object":
+				return Array.isArray(flagValue) ?
+					flagValue.some(value => SideEffectsFlagPlugin.moduleHasSideEffects(modulePath, value)) :
+					Object.keys(flagValue).some(key => key === modulePath &&
+						SideEffectsFlagPlugin.moduleHasSideEffects(modulePath, flagValue[key])
+					);
+		}
+	}
```

Comment:
I try to keep it as simple as possible, because this increases the change of other bundles adding support for this too.